src/storage-client/src/source/persist_source.rs

```diff
@@ -340,6 +393,63 @@ fn downcast_stats<'a, T: Data>(stats: &'a dyn DynStats) -> Option<&'a T::Stats>
     }
 }
 
+impl PersistSourceDataStatsImpl<'_> {
+    fn col_json<'a>(&'a self, idx: usize, _arena: &'a RowArena) -> Option<ResultSpec<'a>> {
+        let name = self.desc.0.get_name(idx);
```

Comment:
i have a preference that this instead match on the `ColumnType`, which would let us be loud about unexpected stats types in the future, but feel free to ignore this for this PR if you like, and we can circle back later

Reply:
What are the odds that we'd ever track JSON stats for non-bytes columns? If small I'd prefer to avoid having to maintain the exhaustive match. (If non-small, agreed!)

Reply:
hmm, not sure if I got my point across :). it feels like an error to me (and we probably should make it at least a debug_assert) for a caller of `StructStats::col` to request stats for a column using a type other than that column's type. for example (but without loss on generality), the error path for that method has a `format!` in it, which is reasonable to run in a true error case, but not reasonable to run in the common case